Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -70,52 +70,55 @@ void uartStyleDecoder::serialDecode(double baudRate)
         }
         else
         {
-            uartTransmitting = (uart_bit == 1) ? false : true;  // Uart starts transmitting after start bit (logic low).
+            uartTransmitting = uart_bit != 1;  // Uart starts transmitting after start bit (logic low).
             jitterCompensationNeeded = true;
         }
+
         // Update the pointer, accounting for jitter
         updateSerialPtr(baudRate, uart_bit);
         // Calculate stopping condition
         dist_seconds = (double)serialDistance()/(parent->m_sampleRate_bit);
     }
 
     //Not a single stop bit, or idle bit, in the whole stream.  Wire must be disconnected.
-    if(allZeroes){
+    if (allZeroes)
+	{
         qDebug() << "Wire Disconnect detected!";
         wireDisconnected(parent->m_channel);
         parent->m_isDecoding = false;
-        updateTimer->stop();
+        m_updateTimer.stop();
     }
 }
 
-int uartStyleDecoder::serialDistance()
+int uartStyleDecoder::serialDistance() const
 {
     int back_bit = parent->m_back * 8;
     int bufferEnd_bit = (parent->m_bufferLen-1) * 8;
-    if(back_bit >= serialPtr_bit){
+    if (back_bit >= serialPtr_bit)
         return back_bit - serialPtr_bit;
-    }else return bufferEnd_bit - serialPtr_bit + back_bit;
+    else
+		return bufferEnd_bit - serialPtr_bit + back_bit;
 }
 
 void uartStyleDecoder::updateSerialPtr(double baudRate, unsigned char current_bit)
 {
-    if(jitterCompensationNeeded && uartTransmitting){
+    if (jitterCompensationNeeded && uartTransmitting)
         jitterCompensationNeeded = jitterCompensationProcedure(baudRate, current_bit);
-    }
 
     int distance_between_bits = (parent->m_sampleRate_bit)/baudRate;
-    if(uartTransmitting){
+    if (uartTransmitting)
         serialPtr_bit += distance_between_bits;
-    } else serialPtr_bit += (distance_between_bits - 1);  //Less than one baud period so that it will always see that start bit.
+	else
+		serialPtr_bit += (distance_between_bits - 1);  //Less than one baud period so that it will always see that start bit.
 
-    if (serialPtr_bit >= (parent->m_bufferLen * 8)){
+    if (serialPtr_bit >= (parent->m_bufferLen * 8))
         serialPtr_bit -= (parent->m_bufferLen * 8);
-    }
 }
 
-unsigned char uartStyleDecoder::getNextUartBit(){
-    int coord_byte = serialPtr_bit/8;
-    int coord_bit = serialPtr_bit - (8*coord_byte);
+unsigned char uartStyleDecoder::getUartBit(int bitIndex) const
```

Comment:
I understand the intention here, but I think this function makes sense more in its original form.  This whole class is basically a monolithic state machine that decodes a (virtual representation of a) UART stream, so random access to the buffer doesn't quite make sense.  In real life, the bits from a UART stream would be buffered in a FIFO and correct symbol decoding depends on getting them in this order.

Reply:
Got it. I just read up on what UART is, I see your point